langchain/vectorstores/supabase.py

```diff
@@ -139,17 +141,32 @@ def similarity_search_by_vector_with_relevance_scores(
         match_documents_params = dict(query_embedding=query, match_count=k)
         res = self._client.rpc(self.query_name, match_documents_params).execute()
 
-        match_result = [
-            (
-                Document(
-                    metadata=search.get("metadata", {}),  # type: ignore
-                    page_content=search.get("content", ""),
-                ),
-                search.get("similarity", 0.0),
-            )
-            for search in res.data
-            if search.get("content")
-        ]
+        if self.metadatas == None:
+            match_result = [
+                (
+                    Document(
+                        metadata=search.get("metadata", {}),  # type: ignore
+                        page_content=search.get("content", ""),
+                    ),
+                    search.get("similarity", 0.0),
+                )
+                for search in res.data
+                if search.get("content")
+            ]
+        else:
+            match_result = [
+                (
+                    Document(
+                        metadata=search.get("metadata", {}),  # type: ignore
+                        page_content=search.get("content", ""),
+                    ),
+                    search.get("similarity", 0.0),
+                )
+                for search in res.data
+                for metadata in self.metadatas
+                for key in metadata
+                if search.get("metadata")[key] == metadata[key]
```

Comment:
right now this does the filtering client side - is there no way to do during query?

also for readable probably better to do:
```
match_result = [
    (
        Document(
            metadata=search.get("metadata", {}),  # type: ignore
            page_content=search.get("content", ""),
        ),
        search.get("similarity", 0.0),
    )
    for search in res.data
    if search.get("content")
]
if metadatas:
   ...filter
```

Reply:
Perhaps, it seems like it can be done in LangchainJS using a [SupabaseMetadata filter](https://js.langchain.com/docs/api/vectorstores_supabase/interfaces/SupabaseLibArgs#filter).

I don't think there is any `SupabaseLibArgs` equivalent in Langchain and I'm not quite sure how it works in the JS version. It seems like @fishmeister09 has used it in #5043 and it looks like it requires a custom Supabase function. It would be great if he could share his `match_documents_with_filters` query, might help me understand how it works.

Reply:
#905

Reply:
Yes it seems like `similarity_search()` in Python also accepts filters similar to the JS version but I'm not sure there is a way to add filters when using Supabase as a retriever in, for example, RetrievalQA?

Reply:
In js, you can pass in a second argument as filters
`asRetriever(k?: number, filter?: this["FilterType"]): VectorStoreRetriever<this>;`

so you may remove the filter from here
`const vectorStore = await SupabaseVectorStore.fromExistingIndex(
  new OpenAIEmbeddings(),
  {
    client: SUPABASE_CLIENT,
    tableName: 'documents',
    queryName: 'match_documents_with_filters',
    // filter: { email: 'abc@gmail.com' },
  }
);`
and add it here
`vectorStore.asRetriever(5, { email: 'xyz@gmail.com' }),`

Reply:
Yes, but I believe there is nothing equivalent to this in the Python version of Langchain. 

Is the filtering also done client side in js when using this method?